### ISSUE TITLE
Confusion about the interaction between `gradient_accumulation_steps` and `lr_scheduler`

### ISSUE THREAD

**Issue by KyleYueye** [huggingface/diffusers#4387]
I'm running code from `diffusers/examples/controlnet/train_controlnet.py`, and I'm confused about some interactions between `gradient_accumulation_steps` and `lr_scheduler`
Here's the code
```
  # Scheduler and math around the number of training steps.
  overrode_max_train_steps = False
  num_update_steps_per_epoch = math.ceil(len(train_dataloader) / args.gradient_accumulation_steps)   # args.gradient_accumulation_steps=4
  if args.max_train_steps is None:   # args.max_train_steps=None
      args.max_train_steps = args.num_train_epochs * num_update_steps_per_epoch
      overrode_max_train_steps = True

  lr_scheduler = get_scheduler(
      args.lr_scheduler,
      optimizer=optimizer,
      num_warmup_steps=args.lr_warmup_steps * args.gradient_accumulation_steps,  # args.lr_warmup_steps=300
      num_training_steps=args.max_train_steps * args.gradient_accumulation_steps,
      num_cycles=args.lr_num_cycles,
      power=args.lr_power,
  )
```
Below are some parameters about training and tensorboard records. 
![1690798044655](https://github.com/huggingface/diffusers/assets/34768573/583c2d5a-835a-4c7a-8e85-0ba6c6af4e03)
![1690798114092](https://github.com/huggingface/diffusers/assets/34768573/30449725-3078-473a-85d9-059d1c75f43b)
As you can see, the scheduler cannot decrease the learning rate to zero in 4854 steps. Appearantly it *will* be zero in 4854*4 steps, which cannot be reached.  (`args.max_train_steps * args.gradient_accumulation_steps`)
Thus my question is whether the argument `num_training_steps` of `get_scheduler()` should multiply `args.gradient_accumulation_steps`?

**Comment by sayakpaul** [comment]
Was discussed here: https://github.com/huggingface/diffusers/issues/3954. 

Also cc: @eliphatfs.

**Comment by eliphatfs** [comment]
The problem has been fixed days ago and now the lines in that example script is:
```python
    lr_scheduler = get_scheduler(
        args.lr_scheduler,
        optimizer=optimizer,
        num_warmup_steps=args.lr_warmup_steps * accelerator.num_processes,
        num_training_steps=args.max_train_steps * accelerator.num_processes,
        num_cycles=args.lr_num_cycles,
        power=args.lr_power,
    )
```
:)

**Comment by sayakpaul** [comment]
Closing this then. Feel free to re-open.
